[windows] Exception: Bad UTF-8 encoding when running flutter doctor or building/running for Windows

<!-- Thank you for using Flutter!

     If you are looking for support, please check out our documentation
     or consider asking a question on Stack Overflow:
      * https://flutter.dev/
      * https://api.flutter.dev/
      * https://stackoverflow.com/questions/tagged/flutter?sort=frequent

     If you have found a bug or if our documentation doesn't have an answer
     to what you're looking for, then fill out the template below. Please read
     our guide to filing a bug first: https://flutter.dev/docs/resources/bug-reports
-->

## Steps to Reproduce

<!-- Please tell us exactly how to reproduce the problem you are running into. -->

1. Set your windows locale to german
2. Install the Visual Studio 2022 Build Tools
3. Install the latest flutter beta (beta, 2.13.0-0.2.pre)
4. Run `flutter doctor`


# > [Temporary solution](https://github.com/flutter/flutter/issues/102451#issuecomment-1124651845)

## Logs

<details>
<summary>Logs</summary>

```
flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel beta, 2.13.0-0.2.pre, on Microsoft Windows [Version 10.0.19044.1645], locale de-DE)
[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
[✓] Chrome - develop for the web
[☠] Visual Studio - develop for Windows (the doctor check crashed)
    ✗ Due to an error, the doctor check did not complete. If the error message below is not helpful, please let us know
      about this issue at https://github.com/flutter/flutter/issues.
    ✗ Exception: Bad UTF-8 encoding (U+FFFD; REPLACEMENT CHARACTER) found while decoding string: [
        {
          "instanceId": "a944307e",
          "installDate": "2022-04-23T14:46:01Z",
          "installationName": "VisualStudio/17.1.5+32414.318",
          "installationPath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools",
          "installationVersion": "17.1.32414.318",
          "productId": "Microsoft.VisualStudio.Product.BuildTools",
          "productPath": "C:\\Program Files (x86)\\Microsoft Visual
          Studio\\2022\\BuildTools\\Common7\\Tools\\LaunchDevCmd.bat",
          "state": 4294967295,
          "isComplete": true,
          "isLaunchable": true,
          "isPrerelease": false,
          "isRebootRequired": false,
          "displayName": "Visual Studio Build Tools 2022",
          "description": "Die Visual Studio Build Tools erm�glichen Ihnen die Erstellung nativer und verwalteter
          MSBuild-basierter Anwendungen, ohne dass die Visual Studio-IDE erforderlich ist. Es stehen Optionen zur
          Installation von Visual C++-Compilern und -Bibliotheken, MFC, ATL sowie C++/CLI-Unterst�tzung zur Verf�gung.",
          "channelId": "VisualStudio.17.Release",
          "channelUri": "https://aka.ms/vs/17/release/channel",
          "enginePath": "C:\\Program Files (x86)\\Microsoft Visual
          Studio\\Installer\\resources\\app\\ServiceHub\\Services\\Microsoft.VisualStudio.Setup.Service",
          "installedChannelUri": "https://aka.ms/vs/17/release/channel",
          "releaseNotes": "https://docs.microsoft.com/en-us/visualstudio/releases/2022/release-notes-v17.1#17.1.5",
          "thirdPartyNotices": "https://go.microsoft.com/fwlink/?LinkId=661288",
          "updateDate": "2022-04-23T14:46:01.6479305Z",
          "catalog": {
            "buildBranch": "d17.1",
            "buildVersion": "17.1.32414.318",
            "id": "VisualStudio/17.1.5+32414.318",
            "localBuild": "build-lab",
            "manifestName": "VisualStudio",
            "manifestType": "installer",
            "productDisplayVersion": "17.1.5",
            "productLine": "Dev17",
            "productLineVersion": "2022",
            "productMilestone": "RTW",
            "productMilestoneIsPreRelease": "False",
            "productName": "Visual Studio",
            "productPatchVersion": "5",
            "productPreReleaseMilestoneSuffix": "1.0",
            "productSemanticVersion": "17.1.5+32414.318",
            "requiredEngineVersion": "3.1.2204.3969"
          },
          "properties": {
            "campaignId": "5",
            "channelManifestId": "VisualStudio.17.Release/17.1.5+32414.318",
            "nickname": "",
            "setupEngineFilePath": "C:\\Program Files (x86)\\Microsoft Visual Studio\\Installer\\setup.exe"
          }
        }
      ]
      . The Flutter team would greatly appreciate if you could file a bug explaining exactly what you were doing when
      this happened:
      https://github.com/flutter/flutter/issues/new/choose
      The source bytes were:
      [91, 13, 10, 32, 32, 123, 13, 10, 32, 32, 32, 32, 34, 105, 110, 115, 116, 97, 110, 99, 101, 73, 100, 34, 58, 32,
      34, 97, 57, 52, 52, 51, 48, 55, 101, 34, 44, 13, 10, 32, 32, 32, 32, 34, 105, 110, 115, 116, 97, 108, 108, 68, 97,
      116, 101, 34, 58, 32, 34, 50, 48, 50, 50, 45, 48, 52, 45, 50, 51, 84, 49, 52, 58, 52, 54, 58, 48, 49, 90, 34, 44,
      13, 10, 32, 32, 32, 32, 34, 105, 110, 115, 116, 97, 108, 108, 97, 116, 105, 111, 110, 78, 97, 109, 101, 34, 58,
      32, 34, 86, 105, 115, 117, 97, 108, 83, 116, 117, 100, 105, 111, 47, 49, 55, 46, 49, 46, 53, 43, 51, 50, 52, 49,
      52, 46, 51, 49, 56, 34, 44, 13, 10, 32, 32, 32, 32, 34, 105, 110, 115, 116, 97, 108, 108, 97, 116, 105, 111, 110,
      80, 97, 116, 104, 34, 58, 32, 34, 67, 58, 92, 92, 80, 114, 111, 103, 114, 97, 109, 32, 70, 105, 108, 101, 115, 32,
      40, 120, 56, 54, 41, 92, 92, 77, 105, 99, 114, 111, 115, 111, 102, 116, 32, 86, 105, 115, 117, 97, 108, 32, 83,
      116, 117, 100, 105, 111, 92, 92, 50, 48, 50, 50, 92, 92, 66, 117, 105, 108, 100, 84, 111, 111, 108, 115, 34, 44,
      13, 10, 32, 32, 32, 32, 34, 105, 110, 115, 116, 97, 108, 108, 97, 116, 105, 111, 110, 86, 101, 114, 115, 105, 111,
      110, 34, 58, 32, 34, 49, 55, 46, 49, 46, 51, 50, 52, 49, 52, 46, 51, 49, 56, 34, 44, 13, 10, 32, 32, 32, 32, 34,
      112, 114, 111, 100, 117, 99, 116, 73, 100, 34, 58, 32, 34, 77, 105, 99, 114, 111, 115, 111, 102, 116, 46, 86, 105,
      115, 117, 97, 108, 83, 116, 117, 100, 105, 111, 46, 80, 114, 111, 100, 117, 99, 116, 46, 66, 117, 105, 108, 100,
      84, 111, 111, 108, 115, 34, 44, 13, 10, 32, 32, 32, 32, 34, 112, 114, 111, 100, 117, 99, 116, 80, 97, 116, 104,
      34, 58, 32, 34, 67, 58, 92, 92, 80, 114, 111, 103, 114, 97, 109, 32, 70, 105, 108, 101, 115, 32, 40, 120, 56, 54,
      41, 92, 92, 77, 105, 99, 114, 111, 115, 111, 102, 116, 32, 86, 105, 115, 117, 97, 108, 32, 83, 116, 117, 100, 105,
      111, 92, 92, 50, 48, 50, 50, 92, 92, 66, 117, 105, 108, 100, 84, 111, 111, 108, 115, 92, 92, 67, 111, 109, 109,
      111, 110, 55, 92, 92, 84, 111, 111, 108, 115, 92, 92, 76, 97, 117, 110, 99, 104, 68, 101, 118, 67, 109, 100, 46,
      98, 97, 116, 34, 44, 13, 10, 32, 32, 32, 32, 34, 115, 116, 97, 116, 101, 34, 58, 32, 52, 50, 57, 52, 57, 54, 55,
      50, 57, 53, 44, 13, 10, 32, 32, 32, 32, 34, 105, 115, 67, 111, 109, 112, 108, 101, 116, 101, 34, 58, 32, 116, 114,
      117, 101, 44, 13, 10, 32, 32, 32, 32, 34, 105, 115, 76, 97, 117, 110, 99, 104, 97, 98, 108, 101, 34, 58, 32, 116,
      114, 117, 101, 44, 13, 10, 32, 32, 32, 32, 34, 105, 115, 80, 114, 101, 114, 101, 108, 101, 97, 115, 101, 34, 58,
      32, 102, 97, 108, 115, 101, 44, 13, 10, 32, 32, 32, 32, 34, 105, 115, 82, 101, 98, 111, 111, 116, 82, 101, 113,
      117, 105, 114, 101, 100, 34, 58, 32, 102, 97, 108, 115, 101, 44, 13, 10, 32, 32, 32, 32, 34, 100, 105, 115, 112,
      108, 97, 121, 78, 97, 109, 101, 34, 58, 32, 34, 86, 105, 115, 117, 97, 108, 32, 83, 116, 117, 100, 105, 111, 32,
      66, 117, 105, 108, 100, 32, 84, 111, 111, 108, 115, 32, 50, 48, 50, 50, 34, 44, 13, 10, 32, 32, 32, 32, 34, 100,
      101, 115, 99, 114, 105, 112, 116, 105, 111, 110, 34, 58, 32, 34, 68, 105, 101, 32, 86, 105, 115, 117, 97, 108, 32,
      83, 116, 117, 100, 105, 111, 32, 66, 117, 105, 108, 100, 32, 84, 111, 111, 108, 115, 32, 101, 114, 109, 148, 103,
      108, 105, 99, 104, 101, 110, 32, 73, 104, 110, 101, 110, 32, 100, 105, 101, 32, 69, 114, 115, 116, 101, 108, 108,
      117, 110, 103, 32, 110, 97, 116, 105, 118, 101, 114, 32, 117, 110, 100, 32, 118, 101, 114, 119, 97, 108, 116, 101,
      116, 101, 114, 32, 77, 83, 66, 117, 105, 108, 100, 45, 98, 97, 115, 105, 101, 114, 116, 101, 114, 32, 65, 110,
      119, 101, 110, 100, 117, 110, 103, 101, 110, 44, 32, 111, 104, 110, 101, 32, 100, 97, 115, 115, 32, 100, 105, 101,
      32, 86, 105, 115, 117, 97, 108, 32, 83, 116, 117, 100, 105, 111, 45, 73, 68, 69, 32, 101, 114, 102, 111, 114, 100,
      101, 114, 108, 105, 99, 104, 32, 105, 115, 116, 46, 32, 69, 115, 32, 115, 116, 101, 104, 101, 110, 32, 79, 112,
      116, 105, 111, 110, 101, 110, 32, 122, 117, 114, 32, 73, 110, 115, 116, 97, 108, 108, 97, 116, 105, 111, 110, 32,
      118, 111, 110, 32, 86, 105, 115, 117, 97, 108, 32, 67, 43, 43, 45, 67, 111, 109, 112, 105, 108, 101, 114, 110, 32,
      117, 110, 100, 32, 45, 66, 105, 98, 108, 105, 111, 116, 104, 101, 107, 101, 110, 44, 32, 77, 70, 67, 44, 32, 65,
      84, 76, 32, 115, 111, 119, 105, 101, 32, 67, 43, 43, 47, 67, 76, 73, 45, 85, 110, 116, 101, 114, 115, 116, 129,
      116, 122, 117, 110, 103, 32, 122, 117, 114, 32, 86, 101, 114, 102, 129, 103, 117, 110, 103, 46, 34, 44, 13, 10,
      32, 32, 32, 32, 34, 99, 104, 97, 110, 110, 101, 108, 73, 100, 34, 58, 32, 34, 86, 105, 115, 117, 97, 108, 83, 116,
      117, 100, 105, 111, 46, 49, 55, 46, 82, 101, 108, 101, 97, 115, 101, 34, 44, 13, 10, 32, 32, 32, 32, 34, 99, 104,
      97, 110, 110, 101, 108, 85, 114, 105, 34, 58, 32, 34, 104, 116, 116, 112, 115, 58, 47, 47, 97, 107, 97, 46, 109,
      115, 47, 118, 115, 47, 49, 55, 47, 114, 101, 108, 101, 97, 115, 101, 47, 99, 104, 97, 110, 110, 101, 108, 34, 44,
      13, 10, 32, 32, 32, 32, 34, 101, 110, 103, 105, 110, 101, 80, 97, 116, 104, 34, 58, 32, 34, 67, 58, 92, 92, 80,
      114, 111, 103, 114, 97, 109, 32, 70, 105, 108, 101, 115, 32, 40, 120, 56, 54, 41, 92, 92, 77, 105, 99, 114, 111,
      115, 111, 102, 116, 32, 86, 105, 115, 117, 97, 108, 32, 83, 116, 117, 100, 105, 111, 92, 92, 73, 110, 115, 116,
      97, 108, 108, 101, 114, 92, 92, 114, 101, 115, 111, 117, 114, 99, 101, 115, 92, 92, 97, 112, 112, 92, 92, 83, 101,
      114, 118, 105, 99, 101, 72, 117, 98, 92, 92, 83, 101, 114, 118, 105, 99, 101, 115, 92, 92, 77, 105, 99, 114, 111,
      115, 111, 102, 116, 46, 86, 105, 115, 117, 97, 108, 83, 116, 117, 100, 105, 111, 46, 83, 101, 116, 117, 112, 46,
      83, 101, 114, 118, 105, 99, 101, 34, 44, 13, 10, 32, 32, 32, 32, 34, 105, 110, 115, 116, 97, 108, 108, 101, 100,
      67, 104, 97, 110, 110, 101, 108, 85, 114, 105, 34, 58, 32, 34, 104, 116, 116, 112, 115, 58, 47, 47, 97, 107, 97,
      46, 109, 115, 47, 118, 115, 47, 49, 55, 47, 114, 101, 108, 101, 97, 115, 101, 47, 99, 104, 97, 110, 110, 101, 108,
      34, 44, 13, 10, 32, 32, 32, 32, 34, 114, 101, 108, 101, 97, 115, 101, 78, 111, 116, 101, 115, 34, 58, 32, 34, 104,
      116, 116, 112, 115, 58, 47, 47, 100, 111, 99, 115, 46, 109, 105, 99, 114, 111, 115, 111, 102, 116, 46, 99, 111,
      109, 47, 101, 110, 45, 117, 115, 47, 118, 105, 115, 117, 97, 108, 115, 116, 117, 100, 105, 111, 47, 114, 101, 108,
      101, 97, 115, 101, 115, 47, 50, 48, 50, 50, 47, 114, 101, 108, 101, 97, 115, 101, 45, 110, 111, 116, 101, 115, 45,
      118, 49, 55, 46, 49, 35, 49, 55, 46, 49, 46, 53, 34, 44, 13, 10, 32, 32, 32, 32, 34, 116, 104, 105, 114, 100, 80,
      97, 114, 116, 121, 78, 111, 116, 105, 99, 101, 115, 34, 58, 32, 34, 104, 116, 116, 112, 115, 58, 47, 47, 103, 111,
      46, 109, 105, 99, 114, 111, 115, 111, 102, 116, 46, 99, 111, 109, 47, 102, 119, 108, 105, 110, 107, 47, 63, 76,
      105, 110, 107, 73, 100, 61, 54, 54, 49, 50, 56, 56, 34, 44, 13, 10, 32, 32, 32, 32, 34, 117, 112, 100, 97, 116,
      101, 68, 97, 116, 101, 34, 58, 32, 34, 50, 48, 50, 50, 45, 48, 52, 45, 50, 51, 84, 49, 52, 58, 52, 54, 58, 48, 49,
      46, 54, 52, 55, 57, 51, 48, 53, 90, 34, 44, 13, 10, 32, 32, 32, 32, 34, 99, 97, 116, 97, 108, 111, 103, 34, 58,
      32, 123, 13, 10, 32, 32, 32, 32, 32, 32, 34, 98, 117, 105, 108, 100, 66, 114, 97, 110, 99, 104, 34, 58, 32, 34,
      100, 49, 55, 46, 49, 34, 44, 13, 10, 32, 32, 32, 32, 32, 32, 34, 98, 117, 105, 108, 100, 86, 101, 114, 115, 105,
      111, 110, 34, 58, 32, 34, 49, 55, 46, 49, 46, 51, 50, 52, 49, 52, 46, 51, 49, 56, 34, 44, 13, 10, 32, 32, 32, 32,
      32, 32, 34, 105, 100, 34, 58, 32, 34, 86, 105, 115, 117, 97, 108, 83, 116, 117, 100, 105, 111, 47, 49, 55, 46, 49,
      46, 53, 43, 51, 50, 52, 49, 52, 46, 51, 49, 56, 34, 44, 13, 10, 32, 32, 32, 32, 32, 32, 34, 108, 111, 99, 97, 108,
      66, 117, 105, 108, 100, 34, 58, 32, 34, 98, 117, 105, 108, 100, 45, 108, 97, 98, 34, 44, 13, 10, 32, 32, 32, 32,
      32, 32, 34, 109, 97, 110, 105, 102, 101, 115, 116, 78, 97, 109, 101, 34, 58, 32, 34, 86, 105, 115, 117, 97, 108,
      83, 116, 117, 100, 105, 111, 34, 44, 13, 10, 32, 32, 32, 32, 32, 32, 34, 109, 97, 110, 105, 102, 101, 115, 116,
      84, 121, 112, 101, 34, 58, 32, 34, 105, 110, 115, 116, 97, 108, 108, 101, 114, 34, 44, 13, 10, 32, 32, 32, 32, 32,
      32, 34, 112, 114, 111, 100, 117, 99, 116, 68, 105, 115, 112, 108, 97, 121, 86, 101, 114, 115, 105, 111, 110, 34,
      58, 32, 34, 49, 55, 46, 49, 46, 53, 34, 44, 13, 10, 32, 32, 32, 32, 32, 32, 34, 112, 114, 111, 100, 117, 99, 116,
      76, 105, 110, 101, 34, 58, 32, 34, 68, 101, 118, 49, 55, 34, 44, 13, 10, 32, 32, 32, 32, 32, 32, 34, 112, 114,
      111, 100, 117, 99, 116, 76, 105, 110, 101, 86, 101, 114, 115, 105, 111, 110, 34, 58, 32, 34, 50, 48, 50, 50, 34,
      44, 13, 10, 32, 32, 32, 32, 32, 32, 34, 112, 114, 111, 100, 117, 99, 116, 77, 105, 108, 101, 115, 116, 111, 110,
      101, 34, 58, 32, 34, 82, 84, 87, 34, 44, 13, 10, 32, 32, 32, 32, 32, 32, 34, 112, 114, 111, 100, 117, 99, 116, 77,
      105, 108, 101, 115, 116, 111, 110, 101, 73, 115, 80, 114, 101, 82, 101, 108, 101, 97, 115, 101, 34, 58, 32, 34,
      70, 97, 108, 115, 101, 34, 44, 13, 10, 32, 32, 32, 32, 32, 32, 34, 112, 114, 111, 100, 117, 99, 116, 78, 97, 109,
      101, 34, 58, 32, 34, 86, 105, 115, 117, 97, 108, 32, 83, 116, 117, 100, 105, 111, 34, 44, 13, 10, 32, 32, 32, 32,
      32, 32, 34, 112, 114, 111, 100, 117, 99, 116, 80, 97, 116, 99, 104, 86, 101, 114, 115, 105, 111, 110, 34, 58, 32,
      34, 53, 34, 44, 13, 10, 32, 32, 32, 32, 32, 32, 34, 112, 114, 111, 100, 117, 99, 116, 80, 114, 101, 82, 101, 108,
      101, 97, 115, 101, 77, 105, 108, 101, 115, 116, 111, 110, 101, 83, 117, 102, 102, 105, 120, 34, 58, 32, 34, 49,
      46, 48, 34, 44, 13, 10, 32, 32, 32, 32, 32, 32, 34, 112, 114, 111, 100, 117, 99, 116, 83, 101, 109, 97, 110, 116,
      105, 99, 86, 101, 114, 115, 105, 111, 110, 34, 58, 32, 34, 49, 55, 46, 49, 46, 53, 43, 51, 50, 52, 49, 52, 46, 51,
      49, 56, 34, 44, 13, 10, 32, 32, 32, 32, 32, 32, 34, 114, 101, 113, 117, 105, 114, 101, 100, 69, 110, 103, 105,
      110, 101, 86, 101, 114, 115, 105, 111, 110, 34, 58, 32, 34, 51, 46, 49, 46, 50, 50, 48, 52, 46, 51, 57, 54, 57,
      34, 13, 10, 32, 32, 32, 32, 125, 44, 13, 10, 32, 32, 32, 32, 34, 112, 114, 111, 112, 101, 114, 116, 105, 101, 115,
      34, 58, 32, 123, 13, 10, 32, 32, 32, 32, 32, 32, 34, 99, 97, 109, 112, 97, 105, 103, 110, 73, 100, 34, 58, 32, 34,
      53, 34, 44, 13, 10, 32, 32, 32, 32, 32, 32, 34, 99, 104, 97, 110, 110, 101, 108, 77, 97, 110, 105, 102, 101, 115,
      116, 73, 100, 34, 58, 32, 34, 86, 105, 115, 117, 97, 108, 83, 116, 117, 100, 105, 111, 46, 49, 55, 46, 82, 101,
      108, 101, 97, 115, 101, 47, 49, 55, 46, 49, 46, 53, 43, 51, 50, 52, 49, 52, 46, 51, 49, 56, 34, 44, 13, 10, 32,
      32, 32, 32, 32, 32, 34, 110, 105, 99, 107, 110, 97, 109, 101, 34, 58, 32, 34, 34, 44, 13, 10, 32, 32, 32, 32, 32,
      32, 34, 115, 101, 116, 117, 112, 69, 110, 103, 105, 110, 101, 70, 105, 108, 101, 80, 97, 116, 104, 34, 58, 32, 34,
      67, 58, 92, 92, 80, 114, 111, 103, 114, 97, 109, 32, 70, 105, 108, 101, 115, 32, 40, 120, 56, 54, 41, 92, 92, 77,
      105, 99, 114, 111, 115, 111, 102, 116, 32, 86, 105, 115, 117, 97, 108, 32, 83, 116, 117, 100, 105, 111, 92, 92,
      73, 110, 115, 116, 97, 108, 108, 101, 114, 92, 92, 115, 101, 116, 117, 112, 46, 101, 120, 101, 34, 13, 10, 32, 32,
      32, 32, 125, 13, 10, 32, 32, 125, 13, 10, 93, 13, 10]


[✓] Android Studio (version 2021.1)
[✓] IntelliJ IDEA Ultimate Edition (version 2022.1)
[✓] VS Code (version 1.66.2)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

! Doctor found issues in 1 category.

```
</details>

> 临时解决方案：将`vswhere.exe`替换`C:\Program Files (x86)\Microsoft Visual Studio\Installer`为最新版本：https://github.com/microsoft/vswhere/releases
The following error occurred after I replaced

CMake Error at CMakeLists.txt:2 (project):
  Generator

    Visual Studio 17 2022

  could not find specified instance of Visual Studio:

    D:/visualStudio

--- a/packages/flutter_tools/lib/src/convert.dart
+++ b/packages/flutter_tools/lib/src/convert.dart
@@ -26,10 +26,14 @@ const Encoding utf8ForTesting = cnv.utf8;
 /// that aren't UTF-8 and we're not quite sure how this is happening.
 /// This tells people to report a bug when they see this.
 class Utf8Codec extends Encoding {
-  const Utf8Codec();
+  const Utf8Codec({this.reportErrors = true});
+
+  final bool reportErrors;
 
   @override
-  Converter<List<int>, String> get decoder => const Utf8Decoder();
+  Converter<List<int>, String> get decoder => reportErrors
+    ? const Utf8Decoder()
+    : const Utf8Decoder(reportErrors: false);
 
   @override
   Converter<String, List<int>> get encoder => cnv.utf8.encoder;

--- a/packages/flutter_tools/lib/src/windows/visual_studio.dart
+++ b/packages/flutter_tools/lib/src/windows/visual_studio.dart
@@ -52,7 +52,8 @@ class VisualStudio {
 
   /// The user-friendly version number of the Visual Studio install.
   ///
-  /// For instance: "15.4.0".
+  /// For instance: "15.4.0". This should only be used for display purposes.
+  /// Logic based off the installation's version should use the `fullVersion`.
   String? get displayVersion => _bestVisualStudioDetails?.catalogDisplayVersion;
 
   /// The directory where Visual Studio is installed.
@@ -419,21 +420,29 @@ class VswhereDetails {
 
     return VswhereDetails(
       meetsRequirements: meetsRequirements,
-      installationPath: _validateString(details['installationPath'] as String?),
-      displayName: _validateString(details['displayName'] as String?),
-      fullVersion: _validateString(details['installationVersion'] as String?),
       isComplete: details['isComplete'] as bool?,
       isLaunchable: details['isLaunchable'] as bool?,
       isRebootRequired: details['isRebootRequired'] as bool?,
       isPrerelease: details['isPrerelease'] as bool?,
-      catalogDisplayVersion: catalog == null ? null : _validateString(catalog['productDisplayVersion'] as String?),
+
+      // Below are strings that must be well-formed without replacement characters.
+      installationPath: _validateString(details['installationPath'] as String?),
+      displayName: _validateString(details['displayName'] as String?),
+      fullVersion: _validateString(details['installationVersion'] as String?),
+
+      // Below are strings that are used only for display purposes and are allowed to
+      // contain replacement characters.
+      catalogDisplayVersion: catalog == null ? null : catalog['productDisplayVersion'] as String?,
     );
   }
 
+  /// Verify JSON strings from vswhere.exe output are valid.
+  ///
+  /// The output of vswhere.exe is known to output replacement characters.
+  /// Use this to ensure values that must be well-formed are valid. Strings that
+  /// are only used for display purposes should skip this check.
+  /// See: https://github.com/flutter/flutter/issues/102451
   static String? _validateString(String? value) {
-    // The output of vswhere.exe is known to output replacement characters.
-    // Ensure this does not affect values used by Flutter.
-    // See: https://github.com/flutter/flutter/issues/102451
     if (value != null && value.contains('\u{FFFD}')) {
       throwToolExit(
         'Bad UTF-8 encoding (U+FFFD; REPLACEMENT CHARACTER) found in string: $value. '

--- a/packages/flutter_tools/lib/src/windows/visual_studio.dart
+++ b/packages/flutter_tools/lib/src/windows/visual_studio.dart
@@ -47,7 +47,8 @@ class VisualStudio {
 
   /// The name of the Visual Studio install.
   ///
-  /// For instance: "Visual Studio Community 2019".
+  /// For instance: "Visual Studio Community 2019". This should only be used for
+  /// display purposes.
   String? get displayName => _bestVisualStudioDetails?.displayName;
 
   /// The user-friendly version number of the Visual Studio install.
@@ -427,11 +428,11 @@ class VswhereDetails {
 
       // Below are strings that must be well-formed without replacement characters.
       installationPath: _validateString(details['installationPath'] as String?),
-      displayName: _validateString(details['displayName'] as String?),
       fullVersion: _validateString(details['installationVersion'] as String?),
 
       // Below are strings that are used only for display purposes and are allowed to
       // contain replacement characters.
+      displayName: details['displayName'] as String?,
       catalogDisplayVersion: catalog == null ? null : catalog['productDisplayVersion'] as String?,
     );
   }

--- a/packages/flutter_tools/lib/src/windows/visual_studio.dart
+++ b/packages/flutter_tools/lib/src/windows/visual_studio.dart
@@ -2,6 +2,8 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:convert';
+
 import 'package:meta/meta.dart';
 import 'package:process/process.dart';
 
@@ -12,7 +14,6 @@ import '../base/logger.dart';
 import '../base/platform.dart';
 import '../base/process.dart';
 import '../base/version.dart';
-import '../convert.dart';
 
 /// Encapsulates information about the installed copy of Visual Studio, if any.
 class VisualStudio {
@@ -282,6 +283,10 @@ class VisualStudio {
         '-utf8',
         '-latest',
       ];
+      // Ideally this would use Flutter's utf8 encoding which errors on malformed UTF-8.
+      // However, vswhere.exe is known to output unicode replacement characters.
+      // These replacement characters will be ignored unless they affect used properties.
+      // See: https://github.com/flutter/flutter/issues/102451
       final RunResult whereResult = _processUtils.runSync(<String>[
         _vswherePath,
         ...defaultArguments,
@@ -416,15 +421,31 @@ class VswhereDetails {
 
     return VswhereDetails(
       meetsRequirements: meetsRequirements,
-      installationPath: details['installationPath'] as String?,
-      displayName: details['displayName'] as String?,
-      fullVersion: details['installationVersion'] as String?,
+      installationPath: _readString(details['installationPath']),
+      displayName: _readString(details['displayName']),
+      fullVersion: _readString(details['installationVersion']),
       isComplete: details['isComplete'] as bool?,
       isLaunchable: details['isLaunchable'] as bool?,
       isRebootRequired: details['isRebootRequired'] as bool?,
       isPrerelease: details['isPrerelease'] as bool?,
-      catalogDisplayVersion: catalog == null ? null : catalog['productDisplayVersion'] as String?,
+      catalogDisplayVersion: catalog == null ? null : _readString(catalog['productDisplayVersion']),
     );
+  }
+
+  static String? _readString(dynamic value) {
+    // The output of vswhere.exe is known to output replacement characters.
+    // Ensure this does not affect values used by Flutter.
+    // See: https://github.com/flutter/flutter/issues/102451
+    final String? stringValue = value as String?;
+    if (value != null && value.contains('\u{FFFD}')) {
+      throwToolExit(
+        'Bad UTF-8 encoding (U+FFFD; REPLACEMENT CHARACTER) found in string: $value. '
+        'The Flutter team would greatly appreciate if you could file a bug explaining '
+        'exactly what you were doing when this happened:\n'
+        'https://github.com/flutter/flutter/issues/new/choose\n');
+    }
+
+    return stringValue;
   }
 
   /// Whether the installation satisfies the required workloads and minimum version.

--- a/packages/flutter_tools/lib/src/windows/visual_studio.dart
+++ b/packages/flutter_tools/lib/src/windows/visual_studio.dart
@@ -282,16 +282,15 @@ class VisualStudio {
         '-utf8',
         '-latest',
       ];
-      // Ideally this would use Flutter's utf8 encoding which errors on malformed UTF-8.
-      // However, vswhere.exe is known to output unicode replacement characters.
-      // These replacement characters will be ignored unless they affect used properties.
+      // Ignore replacement characters as vswhere.exe is known to output them.
       // See: https://github.com/flutter/flutter/issues/102451
+      const Encoding encoding = Utf8Codec(reportErrors: false);
       final RunResult whereResult = _processUtils.runSync(<String>[
         _vswherePath,
         ...defaultArguments,
         ...?additionalArguments,
         ...requirementArguments,
-      ], encoding: const Utf8Codec(reportErrors: false));
+      ], encoding: encoding);
       if (whereResult.exitCode == 0) {
         final List<Map<String, dynamic>> installations =
             (json.decode(whereResult.stdout) as List<dynamic>).cast<Map<String, dynamic>>();

--- a/packages/flutter_tools/lib/src/windows/visual_studio.dart
+++ b/packages/flutter_tools/lib/src/windows/visual_studio.dart
@@ -2,8 +2,6 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-import 'dart:convert';
-
 import 'package:meta/meta.dart';
 import 'package:process/process.dart';
 
@@ -14,6 +12,7 @@ import '../base/logger.dart';
 import '../base/platform.dart';
 import '../base/process.dart';
 import '../base/version.dart';
+import '../convert.dart';
 
 /// Encapsulates information about the installed copy of Visual Studio, if any.
 class VisualStudio {
@@ -292,7 +291,7 @@ class VisualStudio {
         ...defaultArguments,
         ...?additionalArguments,
         ...requirementArguments,
-      ], encoding: utf8);
+      ], encoding: const Utf8Codec(reportErrors: false));
       if (whereResult.exitCode == 0) {
         final List<Map<String, dynamic>> installations =
             (json.decode(whereResult.stdout) as List<dynamic>).cast<Map<String, dynamic>>();
@@ -421,22 +420,21 @@ class VswhereDetails {
 
     return VswhereDetails(
       meetsRequirements: meetsRequirements,
-      installationPath: _readString(details['installationPath']),
-      displayName: _readString(details['displayName']),
-      fullVersion: _readString(details['installationVersion']),
+      installationPath: _validateString(details['installationPath'] as String?),
+      displayName: _validateString(details['displayName'] as String?),
+      fullVersion: _validateString(details['installationVersion'] as String?),
       isComplete: details['isComplete'] as bool?,
       isLaunchable: details['isLaunchable'] as bool?,
       isRebootRequired: details['isRebootRequired'] as bool?,
       isPrerelease: details['isPrerelease'] as bool?,
-      catalogDisplayVersion: catalog == null ? null : _readString(catalog['productDisplayVersion']),
+      catalogDisplayVersion: catalog == null ? null : _validateString(catalog['productDisplayVersion'] as String?),
     );
   }
 
-  static String? _readString(dynamic value) {
+  static String? _validateString(String? value) {
     // The output of vswhere.exe is known to output replacement characters.
     // Ensure this does not affect values used by Flutter.
     // See: https://github.com/flutter/flutter/issues/102451
-    final String? stringValue = value as String?;
     if (value != null && value.contains('\u{FFFD}')) {
       throwToolExit(
         'Bad UTF-8 encoding (U+FFFD; REPLACEMENT CHARACTER) found in string: $value. '
@@ -445,7 +443,7 @@ class VswhereDetails {
         'https://github.com/flutter/flutter/issues/new/choose\n');
     }
 
-    return stringValue;
+    return value;
   }
 
   /// Whether the installation satisfies the required workloads and minimum version.

--- a/packages/flutter_tools/test/general.shard/windows/visual_studio_test.dart
+++ b/packages/flutter_tools/test/general.shard/windows/visual_studio_test.dart
@@ -922,7 +922,7 @@ void main() {
           throwsToolExit(message: 'Bad UTF-8 encoding (U+FFFD; REPLACEMENT CHARACTER) found in string'));
     });
 
-    testWithoutContext("Throws ToolExit on bad UTF-8 in catalog's productDisplayVersion", () {
+    testWithoutContext("Ignores bad UTF-8 in catalog's productDisplayVersion", () {
       final Map<String, dynamic> catalog = Map<String, dynamic>.of(_defaultResponse['catalog'] as Map<String, dynamic>)
         ..['productDisplayVersion'] = '\u{FFFD}';
       final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse)
@@ -930,8 +930,12 @@ void main() {
 
       setMockCompatibleVisualStudioInstallation(response, fixture.fileSystem, fixture.processManager);
 
-      expect(() => visualStudio.isInstalled,
-          throwsToolExit(message: 'Bad UTF-8 encoding (U+FFFD; REPLACEMENT CHARACTER) found in string'));
+      expect(visualStudio.isInstalled, true);
+      expect(visualStudio.isAtLeastMinimumVersion, true);
+      expect(visualStudio.hasNecessaryComponents, true);
+      expect(visualStudio.cmakePath, equals(cmakePath));
+      expect(visualStudio.cmakeGenerator, equals('Visual Studio 16 2019'));
+      expect(visualStudio.displayVersion, '\u{FFFD}');
     });
   });
 

--- a/packages/flutter_tools/test/general.shard/windows/visual_studio_test.dart
+++ b/packages/flutter_tools/test/general.shard/windows/visual_studio_test.dart
@@ -862,42 +862,17 @@ void main() {
 
       expect(visualStudio.getWindows10SDKVersion(), null);
     });
+  });
 
-    testWithoutContext('Reports unicode replacement char in used properties', () {
-      const String badUtf8 = 'Bad UTF8 \u{FFFD}';
-      final List<void Function(Map<String, dynamic>)> propertyOverrides = <void Function(Map<String, dynamic>)> [
-        (Map<String, dynamic> response) => response['installationPath'] = badUtf8,
-        (Map<String, dynamic> response) => response['displayName'] = badUtf8,
-        (Map<String, dynamic> response) => response['installationVersion'] = badUtf8,
-        (Map<String, dynamic> response) {
-          final Map<String, dynamic> catalog = Map<String, dynamic>.of(response['catalog'] as Map<String, dynamic>);
-          response['catalog'] = catalog;
-          catalog['productDisplayVersion'] = badUtf8;
-        }
-      ];
-
-      for (final void Function(Map<String, dynamic>) propertyOverride in propertyOverrides) {
-        final VisualStudioFixture fixture = setUpVisualStudio();
-        final VisualStudio visualStudio = fixture.visualStudio;
-        final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse);
-        propertyOverride(response);
-
-        setMockCompatibleVisualStudioInstallation(
-          response,
-          fixture.fileSystem,
-          fixture.processManager,
-        );
-
-        expect(() => visualStudio.isInstalled,
-             throwsToolExit(message: 'Bad UTF-8 encoding (U+FFFD; REPLACEMENT CHARACTER) found in string'));
-      }
-    });
-
+  // The output of vswhere.exe is known to contain bad UTF8.
+  // See: https://github.com/flutter/flutter/issues/102451
+  group('Correctly handles bad UTF-8 from vswhere.exe output', () {
     testWithoutContext('Ignores unicode replacement char in unused properties', () {
-      final VisualStudioFixture fixture = setUpVisualStudio();
-      final VisualStudio visualStudio = fixture.visualStudio;
       final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse)
         ..['unused'] = 'Bad UTF8 \u{FFFD}';
+
+      final VisualStudioFixture fixture = setUpVisualStudio();
+      final VisualStudio visualStudio = fixture.visualStudio;
 
       setMockCompatibleVisualStudioInstallation(
         response,
@@ -910,6 +885,60 @@ void main() {
       expect(visualStudio.hasNecessaryComponents, true);
       expect(visualStudio.cmakePath, equals(cmakePath));
       expect(visualStudio.cmakeGenerator, equals('Visual Studio 16 2019'));
+    });
+
+    testWithoutContext('Throws ToolExit on bad UTF-8 in installationPath', () {
+      final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse)
+        ..['installationPath'] = '\u{FFFD}';
+
+      final VisualStudioFixture fixture = setUpVisualStudio();
+      final VisualStudio visualStudio = fixture.visualStudio;
+
+      setMockCompatibleVisualStudioInstallation(response, fixture.fileSystem, fixture.processManager);
+
+      expect(() => visualStudio.isInstalled,
+          throwsToolExit(message: 'Bad UTF-8 encoding (U+FFFD; REPLACEMENT CHARACTER) found in string'));
+    });
+
+    testWithoutContext('Throws ToolExit on bad UTF-8 in displayName', () {
+      final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse)
+        ..['displayName'] = '\u{FFFD}';
+
+      final VisualStudioFixture fixture = setUpVisualStudio();
+      final VisualStudio visualStudio = fixture.visualStudio;
+
+      setMockCompatibleVisualStudioInstallation(response, fixture.fileSystem, fixture.processManager);
+
+      expect(() => visualStudio.isInstalled,
+          throwsToolExit(message: 'Bad UTF-8 encoding (U+FFFD; REPLACEMENT CHARACTER) found in string'));
+    });
+
+    testWithoutContext('Throws ToolExit on bad UTF-8 in installationVersion', () {
+      final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse)
+        ..['installationVersion'] = '\u{FFFD}';
+
+      final VisualStudioFixture fixture = setUpVisualStudio();
+      final VisualStudio visualStudio = fixture.visualStudio;
+
+      setMockCompatibleVisualStudioInstallation(response, fixture.fileSystem, fixture.processManager);
+
+      expect(() => visualStudio.isInstalled,
+          throwsToolExit(message: 'Bad UTF-8 encoding (U+FFFD; REPLACEMENT CHARACTER) found in string'));
+    });
+
+    testWithoutContext("Throws ToolExit on bad UTF-8 in catalog's productDisplayVersion", () {
+      final Map<String, dynamic> catalog = Map<String, dynamic>.of(_defaultResponse['catalog'] as Map<String, dynamic>)
+        ..['productDisplayVersion'] = '\u{FFFD}';
+      final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse)
+        ..['catalog'] = catalog;
+
+      final VisualStudioFixture fixture = setUpVisualStudio();
+      final VisualStudio visualStudio = fixture.visualStudio;
+
+      setMockCompatibleVisualStudioInstallation(response, fixture.fileSystem, fixture.processManager);
+
+      expect(() => visualStudio.isInstalled,
+          throwsToolExit(message: 'Bad UTF-8 encoding (U+FFFD; REPLACEMENT CHARACTER) found in string'));
     });
   });
 

--- a/packages/flutter_tools/test/general.shard/windows/visual_studio_test.dart
+++ b/packages/flutter_tools/test/general.shard/windows/visual_studio_test.dart
@@ -862,6 +862,55 @@ void main() {
 
       expect(visualStudio.getWindows10SDKVersion(), null);
     });
+
+    testWithoutContext('Reports unicode replacement char in used properties', () {
+      const String badUtf8 = 'Bad UTF8 \u{FFFD}';
+      final List<void Function(Map<String, dynamic>)> propertyOverrides = <void Function(Map<String, dynamic>)> [
+        (Map<String, dynamic> response) => response['installationPath'] = badUtf8,
+        (Map<String, dynamic> response) => response['displayName'] = badUtf8,
+        (Map<String, dynamic> response) => response['installationVersion'] = badUtf8,
+        (Map<String, dynamic> response) {
+          final Map<String, dynamic> catalog = Map<String, dynamic>.of(response['catalog'] as Map<String, dynamic>);
+          response['catalog'] = catalog;
+          catalog['productDisplayVersion'] = badUtf8;
+        }
+      ];
+
+      for (final void Function(Map<String, dynamic>) propertyOverride in propertyOverrides) {
+        final VisualStudioFixture fixture = setUpVisualStudio();
+        final VisualStudio visualStudio = fixture.visualStudio;
+        final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse);
+        propertyOverride(response);
+
+        setMockCompatibleVisualStudioInstallation(
+          response,
+          fixture.fileSystem,
+          fixture.processManager,
+        );
+
+        expect(() => visualStudio.isInstalled,
+             throwsToolExit(message: 'Bad UTF-8 encoding (U+FFFD; REPLACEMENT CHARACTER) found in string'));
+      }
+    });
+
+    testWithoutContext('Ignores unicode replacement char in unused properties', () {
+      final VisualStudioFixture fixture = setUpVisualStudio();
+      final VisualStudio visualStudio = fixture.visualStudio;
+      final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse)
+        ..['unused'] = 'Bad UTF8 \u{FFFD}';
+
+      setMockCompatibleVisualStudioInstallation(
+        response,
+        fixture.fileSystem,
+        fixture.processManager,
+      );
+
+      expect(visualStudio.isInstalled, true);
+      expect(visualStudio.isAtLeastMinimumVersion, true);
+      expect(visualStudio.hasNecessaryComponents, true);
+      expect(visualStudio.cmakePath, equals(cmakePath));
+      expect(visualStudio.cmakeGenerator, equals('Visual Studio 16 2019'));
+    });
   });
 
   group(VswhereDetails, () {

--- a/packages/flutter_tools/test/general.shard/windows/visual_studio_test.dart
+++ b/packages/flutter_tools/test/general.shard/windows/visual_studio_test.dart
@@ -902,16 +902,6 @@ void main() {
           throwsToolExit(message: 'Bad UTF-8 encoding (U+FFFD; REPLACEMENT CHARACTER) found in string'));
     });
 
-    testWithoutContext('Throws ToolExit on bad UTF-8 in displayName', () {
-      final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse)
-        ..['displayName'] = '\u{FFFD}';
-
-      setMockCompatibleVisualStudioInstallation(response, fixture.fileSystem, fixture.processManager);
-
-      expect(() => visualStudio.isInstalled,
-          throwsToolExit(message: 'Bad UTF-8 encoding (U+FFFD; REPLACEMENT CHARACTER) found in string'));
-    });
-
     testWithoutContext('Throws ToolExit on bad UTF-8 in installationVersion', () {
       final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse)
         ..['installationVersion'] = '\u{FFFD}';
@@ -920,6 +910,20 @@ void main() {
 
       expect(() => visualStudio.isInstalled,
           throwsToolExit(message: 'Bad UTF-8 encoding (U+FFFD; REPLACEMENT CHARACTER) found in string'));
+    });
+
+    testWithoutContext('Ignores bad UTF-8 in displayName', () {
+      final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse)
+        ..['displayName'] = '\u{FFFD}';
+
+      setMockCompatibleVisualStudioInstallation(response, fixture.fileSystem, fixture.processManager);
+
+      expect(visualStudio.isInstalled, true);
+      expect(visualStudio.isAtLeastMinimumVersion, true);
+      expect(visualStudio.hasNecessaryComponents, true);
+      expect(visualStudio.cmakePath, equals(cmakePath));
+      expect(visualStudio.cmakeGenerator, equals('Visual Studio 16 2019'));
+      expect(visualStudio.displayName, equals('\u{FFFD}'));
     });
 
     testWithoutContext("Ignores bad UTF-8 in catalog's productDisplayVersion", () {
@@ -935,7 +939,7 @@ void main() {
       expect(visualStudio.hasNecessaryComponents, true);
       expect(visualStudio.cmakePath, equals(cmakePath));
       expect(visualStudio.cmakeGenerator, equals('Visual Studio 16 2019'));
-      expect(visualStudio.displayVersion, '\u{FFFD}');
+      expect(visualStudio.displayVersion, equals('\u{FFFD}'));
     });
   });
 

--- a/packages/flutter_tools/test/general.shard/windows/visual_studio_test.dart
+++ b/packages/flutter_tools/test/general.shard/windows/visual_studio_test.dart
@@ -867,12 +867,17 @@ void main() {
   // The output of vswhere.exe is known to contain bad UTF8.
   // See: https://github.com/flutter/flutter/issues/102451
   group('Correctly handles bad UTF-8 from vswhere.exe output', () {
+    late VisualStudioFixture fixture;
+    late VisualStudio visualStudio;
+
+    setUp(() {
+      fixture = setUpVisualStudio();
+      visualStudio = fixture.visualStudio;
+    });
+
     testWithoutContext('Ignores unicode replacement char in unused properties', () {
       final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse)
         ..['unused'] = 'Bad UTF8 \u{FFFD}';
-
-      final VisualStudioFixture fixture = setUpVisualStudio();
-      final VisualStudio visualStudio = fixture.visualStudio;
 
       setMockCompatibleVisualStudioInstallation(
         response,
@@ -891,9 +896,6 @@ void main() {
       final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse)
         ..['installationPath'] = '\u{FFFD}';
 
-      final VisualStudioFixture fixture = setUpVisualStudio();
-      final VisualStudio visualStudio = fixture.visualStudio;
-
       setMockCompatibleVisualStudioInstallation(response, fixture.fileSystem, fixture.processManager);
 
       expect(() => visualStudio.isInstalled,
@@ -904,9 +906,6 @@ void main() {
       final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse)
         ..['displayName'] = '\u{FFFD}';
 
-      final VisualStudioFixture fixture = setUpVisualStudio();
-      final VisualStudio visualStudio = fixture.visualStudio;
-
       setMockCompatibleVisualStudioInstallation(response, fixture.fileSystem, fixture.processManager);
 
       expect(() => visualStudio.isInstalled,
@@ -916,9 +915,6 @@ void main() {
     testWithoutContext('Throws ToolExit on bad UTF-8 in installationVersion', () {
       final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse)
         ..['installationVersion'] = '\u{FFFD}';
-
-      final VisualStudioFixture fixture = setUpVisualStudio();
-      final VisualStudio visualStudio = fixture.visualStudio;
 
       setMockCompatibleVisualStudioInstallation(response, fixture.fileSystem, fixture.processManager);
 
@@ -931,9 +927,6 @@ void main() {
         ..['productDisplayVersion'] = '\u{FFFD}';
       final Map<String, dynamic> response = Map<String, dynamic>.of(_defaultResponse)
         ..['catalog'] = catalog;
-
-      final VisualStudioFixture fixture = setUpVisualStudio();
-      final VisualStudio visualStudio = fixture.visualStudio;
 
       setMockCompatibleVisualStudioInstallation(response, fixture.fileSystem, fixture.processManager);
 

